src/swap/defi/fantomSonicUpgrade.ts

```diff
@@ -60,21 +60,25 @@ export function makeFantomSonicUpgradePlugin(
 
     const providers = rpcs.map(rpc => new ethers.providers.JsonRpcProvider(rpc))
 
-    const call = async (method: string, args: string[] = []): Promise<any> => {
-      return await asyncWaterfall(
-        shuffleArray(providers).map(provider => async () => {
-          const contract = new ethers.Contract(
-            upgradeContractAddress,
-            ABI,
-            provider
-          )
-          return await (contract[method](...args) as Promise<any>)
-        }),
-        1000
-      )
+    const call = async (method: string): Promise<BigNumber> => {
+      try {
+        return await asyncWaterfall(
+          shuffleArray(providers).map(provider => async () => {
+            const contract = new ethers.Contract(
+              upgradeContractAddress,
+              ABI,
+              provider
+            )
+            return await (contract[method]() as Promise<BigNumber>)
+          }),
+          1000
+        )
+      } catch (error) {
+        throw new Error(`Failed to call method ${method}: ${String(error)}`)
```

Comment:
Should probably mention the plugin this error came from unless we inherit the pluginId or something elsewhere in our output.

Reply:
The core will print the pluginId when reporting the results in the logs